src/rcms/fm/app/level1/HCALxmlHandler.java

```diff
@@ -223,6 +223,10 @@ public String stripExecXML(String execXMLstring, ParameterSet<FunctionManagerPar
       for (StringT maskedApp: maskedAppArray) {
         //logger.info("[JohnLogVector] " + functionManager.FMname + ": about to start masking " + maskedApp.getString());
         String[] maskedAppParts = maskedApp.getString().split("_");
+        if (maskedAppParts.length != 2 || HCALlevelOneEventHandler.stringToInt(maskedAppParts[0]) || !(HCALlevelOneEventHandler.stringToInt(maskedAppParts[1]))){
```

Comment:
I think 2) sort of misses the point of doing the check... iirc the whole reason we got interested in doing this check was because a user put `DTCManager` into a runkey's maskedapps.

The point in the code you point to is within stripExecXML, and function manager related things don't appear in the executive XML. So there is no need for passing the masked FM names to this segment of the code. (Just for Ciaran's sake, the parameter MASKED_RESOURCES contains not only FMs masked using the `maskedFM` attribute of the runkey, but also contains FMs that are masked via the GUI's masking panels.)

There might be a way to check whether the resource that we're stripping the executive XML here corresponds to an FM or an application without making a new parameter. The level1 has a parameter already called AVAILABLE_RESOURCES which is simply a list of the level2 FMs--that could be used, or maybe using the resource's name (`maskedApp` here) to find the qualified resource, and checking its type. There are probably a few ways we could brainstorm to do this.